test/common/ssl/context_impl_test.cc

```diff
@@ -30,6 +30,20 @@ TEST_F(SslContextImplTest, TestdNSNameMatching) {
   EXPECT_FALSE(ContextImpl::dNSNameMatch("lyft.com", ""));
 }
 
+TEST_F(SslContextImplTest, TestURIMatch) {
+  EXPECT_TRUE(ContextImpl::uriMatch("spiffe://lyft.com/foo", "spiffe://lyft.com/foo"));
+  EXPECT_TRUE(ContextImpl::uriMatch("spiffe://lyft.com/*", "spiffe://lyft.com/foo"));
+  EXPECT_TRUE(ContextImpl::uriMatch("spiffe://lyft.com/*", "spiffe://lyft.com/"));
```

Comment:
why not? ;) the first case fall back to `uriPattern == uri`. For the rest two `uri` is compared as a buffer and the prefixes all agree.